server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestCreateIndexAction.java

```diff
@@ -49,6 +54,9 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         if (request.hasContent()) {
             createIndexRequest.source(request.content(), request.getXContentType());
         }
+        if (request.hasParam("update_all_types")) {
+            DEPRECATION_LOGGER.deprecated("[update_all_types] is deprecated since indices may not have more than one type anymore");
```

Comment:
Maybe it would make sense to put this in createIndexRequest#updateAllTypes, so we also log warnings for Java API users, wdyt?